### PR TITLE
Workaround to fix classname issues with storyboards and customModule

### LINKED ISSUE #25: Storyboards: module name for target may not match "customModule" set by Xcode
If a project has 2 targets A and B, and a storyboard file is in both, whenever a scene's custom class is set in Xcode's UI, the `customModule` will also be (automatically) set to the current target's name in the storyboard's XML.

SwiftGen (by default) will preprend the module's name if there is one in the storyboard's XML, so this realisation above means that it will prepend the module name to any custom class, which does not allow generated files to be included by both targets A and B.

### PR DESCRIPTION
**Pull request by jlnquere**
Hey, 

Here is the workaround to fix classname issues with storyboards and customModule (as described in #25) :)

Fixes #25

### CODE REVIEW

File: templates/storyboards-swift3.stencil
Comment (djbe) [inline templates/storyboards-swift3.stencil]: Also, no need to add this newline.

**Review by djbe — CHANGES_REQUESTED** [review]
Please implement this for all storyboard templates, not just the iOS swift3 version.

**Comment by jlnquere** [comment]
Indeed, I forgot the other templates (sorry). Just updated the PR :)

File: templates/storyboards-default.stencil
Comment (djbe) [inline templates/storyboards-default.stencil]: `scene != param.module`
should be
`scene.customModule != param.module`

**Review by djbe — DISMISSED** [review]
Right, so this branch was quite out of date. With the merge, tests are running correctly now.
I noticed an error in the if test when matching `param.module`, it isn't testing the correct property.

Once #2 lands, this PR needs to add the needed documentation for the `ignoreTargetModule` parameter.

I'm wondering if we should add tests for this. @AliSoftware thoughts?

**Comment by AliSoftware** [comment]
Yeah, we totally should test this, specially given how it seems to be some strange behaviour of Xcode with storyboards which might change again in future Xcode version!

**Comment by ldiqual** [comment]
@juli1quere Bump

**Comment by jlnquere** [comment]
In fact, I do not have enough knowledge of Swiftgen to write tests that check this issue :/

**Comment by djbe** [comment]
I'll write some tests if I can find some time this week, no promises though, might be next week.

**Comment by AliSoftware** [comment]
Is this still relevant with the new SwiftGen 5.0 templates? If so, could we finish this PR to include it in the next release?

**Comment by djbe** [comment]
The current templates will always prepend the module to scene classes, even for the target module. In some scenarios, the user might want to disable this, hence this PR.

**Comment by djbe** [comment]
Hmmm, that's annoying... It can't compile our ignore target module files, because it can't find the custom class. The custom class is in an external module, and the import is ignored...

@AliSoftware thoughts? Add something to the filename that says "don't compile this"?

**Comment by AliSoftware** [comment]
Can't we create a stub-env doing the import or something?

**Comment by djbe** [comment]
The thing is, the best way to do this is if we could define the name of the `current` module when checking/compiling the code. At the time, I searched for a bit but didn't really find out how to, any help there would be useful.

If we can define a module when compiling both the output code + the definitions, I think we'll be able to solve this issue by adding the missing class to definitions.swift so that it's both in the "current" module and an imported one.

**Comment by djbe** [comment]
Allright, @AliSoftware no idea what you'll think of this solution, but it works 😄

**Comment by AliSoftware** [comment]
I think at that point basing the decision on the file name is acceptable. But in the long run I hope we won't have more exception and cases to consider.

Otherwise we might need to make our build tests evolve e.g. use a YAML or JSON file to describe the matrix of additional options needed for each template or whatnot. But for now let's go with name matching.

**Comment by ldiqual** [comment]
@djbe Is this ready to go? I see that you removed the WIP label but didn't notify the maintainers.

**Comment by djbe** [comment]
Eh, I left it open in case there was more feedback incoming. I suppose this can get merged 👍

**Comment by ldiqual** [comment]
@AliSoftware Sorry to bother you on this PR, but I was wondering if you had an idea of when this fix will get released?

**Comment by AliSoftware** [comment]
Should be included within the next version of SwiftGen next weekend

**Pull request merged**
